platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
```

Comment:
The better way can be to refactor our Platform class, probably by returning to initial virtual class methods solution and their implementation for each platform. Current approach is faster at runtime but very hard to understand and support.